lib/charms/nginx_ingress_integrator/v0/ingress.py

```diff
@@ -135,12 +136,16 @@ def __init__(self, charm, config_dict):
             if default_key not in config_dict or not config_dict[default_key]:
                 config_dict[default_key] = default_value
 
-        # And now populate data for conformity with charm-relation-interfaces.
+        self.config_dict = self._convert_to_relation_interface(config_dict)
+
+    @staticmethod
+    def _convert_to_relation_interface(config_dict):
+        """create a new relation dict that conforms with charm-relation-interfaces."""
```

Comment:
Add argument and return value description?

Reply:
This is a charm library that will be used by many other charms. So I want to conform with the remaining code in this library and keep changes as minimal as possible.